### PR TITLE
ENH: Add support for resting-state data

### PR DESCRIPTION
**Pull request by dengemann**
@agramfort @hoechenberger second portion

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by hoechenberger** [comment]
I'll add a test case for this before merging.

**Review by agramfort — APPROVED** [review]

File: tests/configs/config_ds000247.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+"""
+MNE Sample Data
```
Comment (hoechenberger) [inline tests/configs/config_ds000247.py:2]: needs fix

File: docs/source/changes.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ authors:
 - Add possibility to exclude runs from the analysis via the new
   [`exclude_runs`][config.exclude_runs] setting.
   ({{ gh(370) }} by {{ authors.crsegerie }})
+- Add possibility to process resting-state data via
+  [`rest_epochs_duration`][config.rest_epochs_duration] and
+  [`rest_epochs_overlap`][config.rest_epochs_overlap].
+  ({{ gh(393) }} by {{ authors.dengemann }})
```
Comment (dengemann) [inline docs/source/changes.md:38]: we should also add Apolline here

Comment (hoechenberger) [reply docs/source/changes.md:38]: I think Merlin contributed too 👍

File: scripts/preprocessing/03-make_epochs.py [hunk 3]
```diff
@@ -52,7 +52,9 @@ def run_epochs(cfg, subject, session=None):
 
     # Generate a unique event name -> event code mapping that can be used
     # across all runs.
-    event_name_to_code_map = config.annotations_to_events(raw_paths=raw_fnames)
+    if cfg.task != "rest":
```
Comment (hoechenberger) [inline scripts/preprocessing/03-make_epochs.py:55]:
```suggestion
    if cfg.task.lower() != 'rest':
```

File: scripts/preprocessing/03-make_epochs.py [hunk 4]
```diff
@@ -63,10 +65,13 @@ def run_epochs(cfg, subject, session=None):
         raw = mne.io.read_raw_fif(raw_fname, preload=True)
 
         # Only keep the subset of the mapping that applies to the current run
-        event_id = event_name_to_code_map.copy()
-        for event_name in event_id.copy().keys():
-            if event_name not in raw.annotations.description:
-                del event_id[event_name]
+        if cfg.task == "rest":
```
Comment (hoechenberger) [inline scripts/preprocessing/03-make_epochs.py:68]:
```suggestion
        if cfg.task.lower() == 'rest':
```

File: tests/run_tests.py [hunk 5]
```diff
@@ -196,6 +204,15 @@ def run_tests(test_suite, download):
         if download:
             fetch(test_options['dataset'])
 
+        # XXX Workaround for buggy date in ds000247. Remove this and the
+        # XXX file referenced here once fixed!!!
+        if dataset == 'ds000247':
+            shutil.copy(
+                src=Path(__file__).parent / 'ds000247_scans.tsv',
+                dst=Path('~/mne_data/ds000247/sub-0002/ses-0001/'
+                        'sub-0002_ses-0001_scans.tsv').expanduser()
```
Comment (hoechenberger) [inline tests/run_tests.py:213]:
```suggestion
                         'sub-0002_ses-0001_scans.tsv').expanduser()
```

**Review by hoechenberger — APPROVED** [review]

File: scripts/sensor/01-make_evoked.py [hunk 6]
```diff
@@ -58,6 +58,10 @@ def run_evoked(cfg, subject, session=None):
             evoked.comment = evoked.comment.replace(orig_cond_name,
                                                     new_cond_name)
             all_evoked[new_cond_name] = evoked
+    elif cfg.conditions is None:
+        # Handle resting-state
+        evoked = epochs.average()
+        all_evoked['rest'] = evoked
```
Comment (agramfort) [inline scripts/sensor/01-make_evoked.py:64]: to me makes no sense to compute the evoked data for a rest task. I would just skip this script for rest

Comment (dengemann) [reply scripts/sensor/01-make_evoked.py:64]: +100

**Comment by hoechenberger** [comment]
Thanks @dengemann @agramfort!! 🥳

**Comment by moritz-gerster** [comment]
Dear all,

Thank you so much for working on this! I would love to use your pipeline for resting-state analyses, especially your great ICA preprocessing steps. But I guess this will not come so soon?

**Pull request merged**